GitUI/UserControls/RevisionGrid/RevisionGridControl.cs

```diff
@@ -2945,9 +2946,26 @@ private void LaunchRebase(string command)
                 return;
             }
 
-            string rebaseCmd = GitCommandHelpers.RebaseCmd(
-                GetActualRevision(LatestSelectedRevision)?.FirstParentId?.ToString(), interactive: true, preserveMerges: false,
-                autosquash: false, autoStash: true, ignoreDate: false, committerDateIsAuthorDate: false, supportRebaseMerges: Module.GitVersion.SupportRebaseMerges);
+            IGitGpgController gpg = new GitGpgController(() => Module, new GPGSecretKeysParser());
+            bool commitSign = false;
+            string defaultKey = null;
+            ThreadHelper.JoinableTaskFactory.Run(async () =>
+            {
+                commitSign = await gpg.GetCommitGPGSignAsync();
+                defaultKey = await gpg.GetDefaultKeyAsync();
+                await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+            });
+
+            string rebaseCmd = GitCommandHelpers.RebaseCmd(GetActualRevision(LatestSelectedRevision)?.FirstParentId?.ToString(),
+                                                           interactive: true,
+                                                           preserveMerges: false,
+                                                           autosquash: false,
+                                                           autoStash: true,
+                                                           ignoreDate: false,
+                                                           committerDateIsAuthorDate: false,
+                                                           supportRebaseMerges: Module.GitVersion.SupportRebaseMerges,
+                                                           gpgSign: commitSign && !string.IsNullOrEmpty(defaultKey),
+                                                           supportNoGpgSign: Module.GitVersion.SupportNoGPGSign);
```

Comment:
This feels very wrong that the revision grid needs to own a GPG controller. This also introduces a non-mockable dependency.